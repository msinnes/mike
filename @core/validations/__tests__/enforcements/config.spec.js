const configEnforcement = require('../../enforcements/config');

describe('configEnforcement', () => {
  it('should be a function', () => {
    expect(configEnforcement).toBeDefined();
    expect(configEnforcement).toBeInstanceOf(Function);
  });

  it('should not throw an error', () => {
    expect(() => {
      configEnforcement();
    }).not.toThrow();
    expect(() => {
      configEnforcement({});
    }).not.toThrow();
    expect(() => {
      configEnforcement({ allowEmpty: true });
    }).not.toThrow();
    expect(() => {
      configEnforcement({ throwOnInvalid: true });
    }).not.toThrow();
    expect(() => {
      configEnforcement({ type: {
        is: () => {},
        message: 'message',
      } });
    }).not.toThrow();
    expect(() => {
      configEnforcement({ allowEmpty: true, throwOnInvalid: true, type: {
        is: () => {},
        message: 'message',
      } });
    }).not.toThrow();
  });

  it('should throw an error', () => {
    expect(() => {
      configEnforcement([]);
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      configEnforcement({ anyOtherKey: true });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      configEnforcement({ allowEmpty: 'not a boolean' });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      configEnforcement({ throwOnInvalid: 'not a boolean' });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      configEnforcement({ type: 'not an object' });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      configEnforcement({ type: {
        notAGoodKey: 'something else',
      } });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      configEnforcement({ type: {
        is: () => {},
      } });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      configEnforcement({ type: {
        message: 'message',
      } });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      configEnforcement({ type: {
        is: 'not a function',
        message: 'message',
      } });
    }).toThrowErrorMatchingSnapshot();
  });
});
