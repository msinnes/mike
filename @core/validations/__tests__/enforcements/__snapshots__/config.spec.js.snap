// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configEnforcement should throw an error 1`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;

exports[`configEnforcement should throw an error 2`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;

exports[`configEnforcement should throw an error 3`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;

exports[`configEnforcement should throw an error 4`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;

exports[`configEnforcement should throw an error 5`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;

exports[`configEnforcement should throw an error 6`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;

exports[`configEnforcement should throw an error 7`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;

exports[`configEnforcement should throw an error 8`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;

exports[`configEnforcement should throw an error 9`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;
